pkg/blob/controllerserver.go

```diff
@@ -781,9 +779,9 @@ func (d *Driver) copyBlobContainer(req *csi.CreateVolumeRequest, accountSasToken
 	case util.AzcopyJobRunning:
 		return fmt.Errorf("wait for the existing AzCopy job to complete, current copy percentage is %s%%", percent)
 	case util.AzcopyJobNotFound:
-		klog.V(2).Infof("copy blob container %s to %s", srcContainerName, dstContainerName)
+		klog.V(2).Infof("copy blob container %s to %s", srcPath, dstPath)
 		execFunc := func() error {
-			cmd := exec.Command("azcopy", "copy", srcPath, dstPath, "--recursive", "--check-length=false")
+			cmd := exec.Command("azcopy", "copy", srcPath+accountSasToken, dstPath+accountSasToken, "--recursive", "--check-length=false")
```

Comment:
this is using the same account sas token, if source account and dest account are different, it would break.

Reply:
should get srcAccountSASToken if it's two accounts, e.g. 
```
	srcAccountSasToken := dstAccountSasToken
	if srcAccountName != dstAccountName && dstAccountSasToken != "" {
		srcAccountOptions := &azure.AccountOptions{
			Name:                srcAccountName,
			ResourceGroup:       accountOptions.ResourceGroup,
			SubscriptionID:      accountOptions.SubscriptionID,
			GetLatestAccountKey: accountOptions.GetLatestAccountKey,
		}
andyzhangx marked this conversation as resolved.
		if srcAccountSasToken, _, err = d.getAzcopyAuth(ctx, srcAccountName, "", storageEndpointSuffix, srcAccountOptions, nil, "", secretNamespace, true); err != nil {
			return err
		}
	}
```

Reply:
I pushed changes which better encapsulate the volume content source validation and copy, including src sas token and associated tests. Please check that `secrets`, `secretName`, `secretNamespace` logic is correct.